test/testdrive/primary-key-optimizations.td

```diff
@@ -122,3 +120,104 @@ $ kafka-ingest format=avro topic=t1 key-format=avro key-schema=${keyschema-2keys
 
 > EXPLAIN SELECT a1.* FROM v2 AS a1, v2 AS a2 WHERE a1.key1 = a2.key1 AND a1.key2 = a2.key2;
 "%0 =| Get v2"
+
+# Declare a key constraint (PRIMARY KEY NOT ENFORCED); otherwise identical tests as above.
```

Comment:
@philip-stoev I copied your tests 1:1 with the modification to manually specify the key. If you think that's overkill or I'm missing some important tests, let me know.

Reply:
@uce: yes the tests are totally good. I am no table to think of anything else. the EXPLAIN plans indicate that the primary key thus defined is considered good as any other.

Reply:
Thanks!